src/lib/qnio/utils.c

```diff
@@ -9,6 +9,117 @@
  */
 
 #include "defs.h"
+#include "qnio.h"
+
+int 
+is_secure()
+{
+    if (access(SECURE_IMPL, F_OK) != 0)
+    {
+        nioDbg("Secure implementation not enabled\n");
+        return 0;
+    }
+    return 1;
+}
+
+SSL_CTX *
+init_server_ssl_ctx()
+{
+    const SSL_METHOD *method;
+    SSL_CTX *ctx = NULL;
+
+    nioDbg("initializing server ssl ctx %s\n", SERVER_KEY);
+    if (access(SERVER_KEY, F_OK) != 0)
+    {
+        nioDbg("Server key not found");
+        return NULL;
+    }
+
+    if (access(SERVER_CERT, F_OK) != 0)
+    {
+        nioDbg("Server cert not found");
+        return NULL;
+    }
+
+    SSL_load_error_strings();   
+    OpenSSL_add_ssl_algorithms();
+    method = SSLv23_server_method();
+
+    ctx = SSL_CTX_new(method);
+    if (!ctx) 
+    {
+        nioDbg("Unable to create SSL context");
+        return NULL;
+    }
+
+    /* Set the key and cert */
+    if (SSL_CTX_use_certificate_file(ctx, SERVER_CERT, SSL_FILETYPE_PEM) < 0) 
+    {
+        nioDbg("Unable to use server certificate");
+        return NULL;
+    }
+
+    if (SSL_CTX_use_PrivateKey_file(ctx, SERVER_KEY, SSL_FILETYPE_PEM) < 0 ) 
+    {
+        nioDbg("Unable to use server key file");
+        return NULL;
+    }
+    return ctx;
+}
+
+SSL_CTX *
+init_client_ssl_ctx(char *instanceid)
+{
+    const SSL_METHOD *method;
+    char clientkey[512] = { 0 };
+    char clientcert[512] = { 0 };
+    SSL_CTX *ctx = NULL;
+
+    strcpy(clientkey, CLIENT_KEYSTORE);
+    strncat(clientkey, instanceid, 64);
+    strncat(clientkey, ".key", 4);
+
+    if (access(clientkey, F_OK) != 0)
+    {
+        nioDbg("Client key not found");
+        return NULL;
+    }
+
+    strcpy(clientcert, CLIENT_KEYSTORE);
+    strncat(clientcert, instanceid, 64);
+    strncat(clientcert, ".cert", 5);
+
```

Comment:
Again, why does QEMU need to be able to provide these paths if libvxhs can reliably read these from a configuration file say /etc/vxhs.conf?
Is the keystore going to change for different clients? 
Regarding the custom CA bundle, this can be an option in /etc/vxhs.conf as well.